com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/DefoldLuaParser.java

```diff
@@ -0,0 +1,189 @@
+package com.dynamo.bob.pipeline;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+
+import com.dynamo.bob.pipeline.luaparser.LuaParser;
+import com.dynamo.bob.pipeline.luaparser.LuaLexer;
+import com.dynamo.bob.pipeline.luaparser.LuaParserBaseListener;
+
+
+public class DefoldLuaParser extends LuaParserBaseListener {
+	private static final Logger LOGGER = Logger.getLogger(DefoldLuaParser.class.getName());
+
+	public class PropertyAndLine {
+		public String property;
+		public int line;
+		public PropertyAndLine(String property, int line) {
+			this.property = property;
+			this.line = line - 1;
+		}
+	}
+
+	private StringBuffer parsedBuffer = null;
+	private CommonTokenStream tokenStream = null;
+
+	private List<String> modules = new ArrayList<String>();
+	private List<PropertyAndLine> properties = new ArrayList<PropertyAndLine>();
+
+	private boolean stripComments = false;
+	private boolean stripProperties = false;
+
+	public DefoldLuaParser() {}
+
+	/**
+	 * Remove Lua comments from the parsed Lua code
+	 * @param strip true if comments should be removed
+	 */
+	public void setStripComments(boolean strip) {
+		stripComments = strip;
+	}
+
+	/**
+	 * Remove go.property() calls from the parsed Lua code
+	 * @param strip true if go.property() calls should be removed
+	 */
+	public void setStripProperties(boolean strip) {
+		stripProperties = strip;
+	}
+
+	/**
+	 * Parse a string containing Lua code. This will detect things such
+	 * as require() and go.property() calls and optionally strip comments and
+	 * go.property() calls.
+	 * @param str Lua code to parse
+	 * @return Parsed string
+	 */
+	public String parse(String str) {
```

Comment:
Main entry point. We pass it some Lua code. The code is parsed by ANTLR and we walk the generated syntax tree in search of require() and go.property() calls.